presto-accumulo/src/main/java/com/facebook/presto/accumulo/conf/AccumuloTableProperties.java

```diff
@@ -0,0 +1,247 @@
+        // This is a comma-delimited list of "presto column:accumulo fam:accumulo qualifier"
+        // triplets
+        Map<String, Pair<String, String>> mapping = new HashMap<>();
+        for (String m : COMMA_SPLITTER.split(strMapping)) {
+            String[] tokens = Iterables.toArray(COLON_SPLITTER.split(m), String.class);
+            checkState(tokens.length == 3, format("Mapping of %s contains %d tokens instead of 3", m, tokens.length));
+            mapping.put(tokens[0], Pair.of(tokens[1], tokens[2]));
+        }
+
+        return mapping;
+    }
+
+    /**
+     * Gets the list of all indexed columns set in the table properties
+     *
+     * @param tableProperties The map of table properties
+     * @return The list of indexed columns, or an empty list if there are none
+     */
+    public static List<String> getIndexColumns(Map<String, Object> tableProperties)
+    {
+        return Arrays.asList(StringUtils.split((String) tableProperties.get(INDEX_COLUMNS), ','));
+    }
+
+    /**
+     * Gets the configured locality groups for the table.
+     *
+     * @param tableProperties The map of table properties
+     * @return Optional map of locality groups
+     */
+    public static Optional<Map<String, Set<String>>> getLocalityGroups(Map<String, Object> tableProperties)
+    {
+        String groupStr = (String) tableProperties.get(LOCALITY_GROUPS);
+        if (groupStr == null) {
+            return Optional.empty();
+        }
+
+        Map<String, Set<String>> groups = new HashMap<>();
+
+        // Split all configured locality groups
+        for (String group : PIPE_SPLITTER.split(groupStr)) {
+            String[] locGroups = Iterables.toArray(COLON_SPLITTER.split(group), String.class);
+
+            if (locGroups.length != 2) {
+                throw new PrestoException(VALIDATION,
+                        "Locality groups string is malformed. See documentation for proper format.");
+            }
+
+            String grpName = locGroups[0];
+            Set<String> colSet = new HashSet<>();
+            groups.put(grpName, colSet);
+
+            for (String f : COMMA_SPLITTER.split(locGroups[1])) {
+                colSet.add(f);
+            }
+        }
+
+        return Optional.of(groups);
+    }
+
+    /**
+     * Gets the configured row ID for the table
+     *
+     * @param tableProperties The map of table properties
+     * @return The row ID, or null if none was specifically set (use the first column)
+     */
+    public static String getRowId(Map<String, Object> tableProperties)
+    {
+        return (String) tableProperties.get(ROW_ID);
+    }
+
+    /**
+     * Gets the scan authorizations, or null if all of the user's scan authorizations should be used
+     *
+     * @param tableProperties The map of table properties
+     * @return The scan authorizations
+     */
+    public static Optional<String> getScanAuthorizations(Map<String, Object> tableProperties)
```

Comment:
Why here `Optional` while in other places `nullable`? E.g. https://github.com/prestodb/presto/pull/5030/files#diff-e0609c76e3698002e0ea2c6334997f55R209
`Optional` is preferred though.